### DISCUSSION TITLE
Working with external components while Unit Testing

### DISCUSSION THREAD

**Discussion by Austen-G** [nasa/fprime#1771] [Q&A] [answered]
Hello! I am current working on writing unit tests for a driver component controlling a UART device (before running on hardware). I would like to test some of the component internals with Fw::Buffers, and was wondering about the use of other components (In this case BufferManager) within unit tests. 

Is the convention in this case simply to mock the output ports of my component to appear as if there is a BufferManager? Or is it to initialize and connect a component like BufferManager for use during the testing. If the latter, how would that be done within the context of the unit test build?

**Comment by timcanham** [comment, marked as answer]
We typically just mock the ports that request buffers. That way you can test cases where it doesn't get one for error handling. It keeps the dependencies of the test just on the component and the test component.
